### PR TITLE
Add more logging around ECID generation and loading (resolves #910)

### LINKED ISSUE #910: Add more logging around ECID generation in Identity
### Prerequisites

- [X] This is not a Security Disclosure, otherwise please follow the guidelines in [Security Policy](https://github.com/adobe/aepsdk-edge-ios/security/policy).
- [X] I have searched in this repository's issues to see if it has already been reported.

### Feature request summary

When Identity extension loads and boots, it should log the current ECID value and where it was retrieved from, either loaded from persistence or generated. This will help debug issues where the ECID is not as expected or changes between launches.

It also might be a good idea to log all key/values loaded from persistence when Identity boots.

### Current behavior

Currently the Identity extension does log when a new ECID value is generated. However, there should also be logging around the value loaded from persistence.

### Expected behavior

_No response_

### Additional implementation details or code snippets

_No response_

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Add logging with ECID value is generated and loaded from persistence. These logs will assist in debugging ECID issues on boot.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPIdentity/Sources/IdentityState.swift [hunk 1]
```diff
@@ -49,6 +49,7 @@ class IdentityState {
 
         // load data from local storage
         identityProperties.loadFromPersistence()
+        Log.debug(label: "\(LOG_TAG):\(#function)", "Load ECID from persistence: ecid = \(identityProperties.ecid?.ecidString ?? "nil")")
```
Comment (emdobrin) [inline AEPIdentity/Sources/IdentityState.swift:52]: nit: can use trace

File: AEPIdentity/Sources/IdentityState.swift [hunk 2]
```diff
@@ -466,9 +467,9 @@ class IdentityState {
     /// Generates the ecid if not cached and save it to persistence
     private func generateAndPersistECID() {
         if identityProperties.ecid == nil {
-            Log.trace(label: "\(LOG_TAG):\(#function)", "Generating new ECID, ECID not found in persistence.")
             identityProperties.ecid = ECID()
             identityProperties.saveToPersistence()
+            Log.trace(label: "\(LOG_TAG):\(#function)", "Generating new ECID value \(identityProperties.ecid?.ecidString ?? "nil")")
```
Comment (emdobrin) [inline AEPIdentity/Sources/IdentityState.swift:472]: can we also add traces on the flow that triggered this call to generateAndPersistECID?

File: AEPIdentity/Sources/IdentityState.swift (hunk 1 above)
Comment (kevinlind) [reply AEPIdentity/Sources/IdentityState.swift]: I've updated this log message to match the one used in Android, which is why it is logging more than previously.

File: AEPIdentity/Sources/IdentityState.swift [hunk 3]
```diff
@@ -244,7 +245,10 @@ class IdentityState {
             Log.trace(label: "\(LOG_TAG):\(#function)", "Not syncing identifiers at this time, no new identifiers or previously synced.")
             syncForIds = false
         } else {
-            generateAndPersistECID()
+            if (identityProperties.ecid == nil) {
```
Comment (emdobrin) [inline AEPIdentity/Sources/IdentityState.swift:248]: I see some new code changes, are these paths covered by tests?

Comment (kevinlind) [reply AEPIdentity/Sources/IdentityState.swift:248]: The nil check is the same as done in `generateAndPersistECID()` so the messages are only logged when the ECID is actually generated.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**